pkg/util/kubernetes/apiserver/services.go

```diff
@@ -0,0 +1,73 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubeapiserver
+
+package apiserver
+
+import (
+	"errors"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/ericchiang/k8s/api/v1"
+)
+
+// mapServices maps each pod (endpoint) to the services connected to it.
+// It is on a per node basis to avoid mixing up the services pods are actually connected to if all pods of different nodes share a similar subnet, therefore sharing a similar IP.
+func (smb *ServiceMapperBundle) mapServices(nodeName string, pods v1.PodList, endpointList v1.EndpointsList) error {
+	smb.m.Lock()
+	defer smb.m.Unlock()
+	ipToEndpoints := make(map[string][]string) // maps the IP address from an endpoint (pod) to associated services ex: "10.10.1.1" : ["service1","service2"]
+	podToIp := make(map[string]string)         // maps the pods of the currently evaluated node to their IP.
+
+	if pods.Items == nil {
+		err := errors.New("empty podlist received")
```

Comment:
What do you think of adding more context in this error, like the `nodeName` or anything relevant ?

Reply:
Indeed, I'm dealing with the empty entrypointlist earlier on, but having a debug for the empty nodename could help.
Although, unless the user has the same subset of IPs on the nodes it would still work properly.